contrib/libpcap/example/main.cc

```diff
@@ -0,0 +1,72 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fcntl.h>
+#include <unistd.h>
+
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "contrib/libpcap/sandboxed.h"
+#include "contrib/libpcap/utils/utils_libpcap.h"
+
+int main(int argc, char* argv[]) {
+  std::string prog_name(argv[0]);
+  google::InitGoogleLogging(argv[0]);
+  std::vector<char*> args = absl::ParseCommandLine(argc, argv);
+
+  if (args.size() != 2 && args.size() != 3) {
+    std::cerr << "Usage " << prog_name << " [PCAP_FILE] [FILTER]\n";
+    return EXIT_FAILURE;
+  }
+
+  LibpcapSapiSandbox sandbox;
+  if (!sandbox.Init().ok()) {
+    std::cerr << "Unable to start sandbox\n";
+    return EXIT_FAILURE;
+  }
+
+  LibPcap libpcap(&sandbox, args[1]);
```

Comment:
I would suggest 1 of the alternative approaches:
 - do not do any initialization work in ctor, instead have `absl::Statsu Init()`
 - use a static factory method that returns StatusOr<LibPcap>

Reply:
This approach was also used in different libraries at this point so I would leave it to be constant with them.

Reply:
Let's make this one nice and clean first, we can fix existing stuff later.